### PR TITLE
Speedup more stable swap math operations

### PR DESCRIPTION
**Pull request by ValarDragon**
Speedup more stableswap math operations. Anticipated 2% speedup to CPU time of syncing

### CODE REVIEW

**Comment by czarcas7ic** [comment]
It seems the failed test is related to the change, will wait to review until passing, unless you want me to look into it @ValarDragon

**Comment by ValarDragon** [comment]
Fixed!

File: osmomath/decimal.go [hunk 1]
```diff
@@ -253,8 +253,8 @@ func (d BigDec) BigInt() *big.Int {
 	return cp.Set(d.i)
 }
 
-// BigIntMut converts BigDec to big.Int, mutative the input
-func (d BigDec) ToBigInt() *big.Int {
+// BigInt returns a copy of the underlying big.Int.
```
Comment (mattverse) [inline osmomath/decimal.go:256]:
```suggestion
// BigIntMut returns the pointer of the underlying big.Int.
```

File: x/gamm/pool-models/stableswap/amm.go [hunk 2]
```diff
@@ -63,15 +63,16 @@ func targetKCalculator(x0, y0, w, yf osmomath.BigDec) osmomath.BigDec {
 // $$k_{iter}(x_f) = -x_{out}^3 + 3 x_0 x_{out}^2 - (y_f^2 + w + 3x_0^2)x_{out}$$
 // where x_out = x_0 - x_f
 func iterKCalculator(x0, w, yf osmomath.BigDec) func(osmomath.BigDec) osmomath.BigDec {
-	// compute coefficients first
-	cubicCoeff := osmomath.OneBigDec().Neg()
+	// compute coefficients first. Notice that the leading coefficient is -1, we will use this to compute faster.
```
Comment (mattverse) [inline x/gamm/pool-models/stableswap/amm.go:66]: note to other reviewers: x_{out}^3 remains same

**Review by mattverse — APPROVED** [review]
ACK 🌮

**Pull request merged**